text/0059-otlp-trace-data-format.md

```diff
@@ -0,0 +1,349 @@
+  // semantic conventions for standardized label names:
+  // https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-semantic-conventions.md
+  repeated AttributeKeyValue labels = 1;
+
+  // dropped_labels_count is the number of dropped labels. If the value is 0, then
+  // no labels were dropped.
+  int32 dropped_labels_count = 2;
+}
+```
+
+### Span
+
+```
+// Span represents a single operation within a trace. Spans can be
+// nested to form a trace tree. Spans may also be linked to other spans
+// from the same or different trace and form graphs. Often, a trace
+// contains a root span that describes the end-to-end latency, and one
+// or more subspans for its sub-operations. A trace can also contain
+// multiple root spans, or none at all. Spans do not need to be
+// contiguous - there may be gaps or overlaps between spans in a trace.
+//
+// The next field id is 18.
+message Span {
+  // trace_id is the unique identifier of a trace. All spans from the same trace share
+  // the same `trace_id`. The ID is a 16-byte array. An ID with all zeroes
+  // is considered invalid.
+  //
+  // This field is semantically required. Receiver should generate new
+  // random trace_id if empty or invalid trace_id was received.
+  //
+  // This field is required.
+  bytes trace_id = 1;
+
+  // span_id is a unique identifier for a span within a trace, assigned when the span
+  // is created. The ID is an 8-byte array. An ID with all zeroes is considered
+  // invalid.
+  //
+  // This field is semantically required. Receiver should generate new
+  // random span_id if empty or invalid span_id was received.
+  //
+  // This field is required.
+  bytes span_id = 2;
+
+  // TraceEntry is the entry that is repeated in tracestate field (see below).
+  message TraceEntry {
+    // key must begin with a lowercase letter, and can only contain
+    // lowercase letters 'a'-'z', digits '0'-'9', underscores '_', dashes
+    // '-', asterisks '*', and forward slashes '/'.
+    string key = 1;
+
+    // value is opaque string up to 256 characters printable ASCII
+    // RFC0020 characters (i.e., the range 0x20 to 0x7E) except ',' and '='.
+    // Note that this also excludes tabs, newlines, carriage returns, etc.
+    string value = 2;
+  }
+
+  // TraceState conveys information about request position in multiple distributed tracing graphs.
+  // It is a list of Tracestate.Entry with a maximum of 32 members in the list.
+  //
+  // See the https://github.com/w3c/distributed-tracing for more details about this field.
+  repeated TraceEntry tracestate = 3;
+
+  // parent_span_id is the `span_id` of this span's parent span. If this is a root span, then this
+  // field must be omitted. The ID is an 8-byte array.
+  bytes parent_span_id = 4;
+
+  // resource that is associated with this span. Optional. If not set, this span
+  // should be part of a ResourceSpans message that does include the resource information,
```

Comment:
I do not see the definition of a `ResourceSpans` message anywhere. I Imaginge it would just be
```protobuf
message ResourceSpans {
    Resource resource = 1;
    repeated Span spans = 2;
}
```

Reply:
It is defined in https://github.com/open-telemetry/oteps/pull/35/files